### PR TITLE
Remove network calls

### LINKED ISSUE #162: Get rid of calls to attach and remove networks to hil (in this version of BMI).
This came up after running HIL/BMI in kumo for a while. 

When user give up their nodes (or we take it forcibly from them), we don't want to deprovision the nodes because that will delete their image, so we just disconnect all the networks and remove it from their project (and with a script, delete the iPXE file).

 Now if a user wants to deprovision a node, they must add that node back to their project, connect the hil_network and only then they can call the `bmi dpro` command.

### PR DESCRIPTION
**Pull request by naved001**
Fixes #162 
Also, fixes the comment issue described in #128 but not any other inconsistency we may have in the CLI.

### CODE REVIEW

**Comment by naved001** [comment]
This one is next on my agenda.

**Comment by naved001** [comment]
@apoorvemohan @Izhmash @mihirborkar 
Let's start reviewing this. I am gonna push a few more commits to this one (or maybe separate PRs if they are unrelated) that will trim down some fat from the BMI code, and then I'll rewrite/split the provision and deprovision call.

**Comment by ianballou** [comment]
This looks good to me so far. Are you still planning to push more code here?

Also, as an aside, how close does this get us to decoupling HIL with M2? I think now the only things that M2 uses HIL for are authentication and reading network/node information.

**Comment by naved001** [comment]
I finished working on this. This only removes the network calls.

I'll removed the call to get Mac information later, and then auth will be the only thing left.

@apoorvemohan can I get another review on this?

**Review by mihirborkar — APPROVED** [review]
LGTM!!! 👍👍

**Comment by ianballou** [comment]
I think it looks good, what about the docs?

**Comment by naved001** [comment]
While updating the docs, I realized that there's no need to pass the nic for deprovisioning anymore (at least in the current implementation).

There are 2 uses of passing the nic I can think of:
1. To disconnect the network from (which won't be needed anymore).
2. To delete the boot configuration file which is named after the mac address. It looks like this:
```
DEFAULT menu
  PROMPT 0
  MENU TITLE PXE Menu
  ONTIMEOUT ubuntu-test


LABEL ubunutu-test
     MENU LABEL ubuntu-image
     KERNEL ipxe.lkrn
     INITRD dell-9.ipxe
```
Right now we don't do 2; does it make sense to do it now? We can either delete the ipxe file, or the boot configuration file or both. @apoorvemohan what do you think?

**Comment by naved001** [comment]
I am gonna leave the nic here as is so that this PR is easier to review. I am gonna make use of nic in deprovisioning later (to delete the boot configuration

**Comment by naved001** [comment]
@Izhmash updated the docs

**Review by ianballou — APPROVED** [review]
I think this looks fine, what's with the CI?

File: ims/cli/cli.py [hunk 1]
```diff
@@ -76,14 +75,11 @@ def provision(project, node, img, network, nic):
     PROJECT = The HIL Project attached to your credentials
     NODE    = The Node to Provision
     IMG     = The Name of the Image to Provision
-    NETWORK = The Name of the Provisioning Network
-    CHANNEL = The Channel to Provision On (For HIL It is 'vlan/native')
-    NIC     = The NIC to use for Network Boot (For HIL IT is 'enp130s0f0')
+    NIC     = The NIC to use for Network Boot
```
Comment (apoorvemohan) [inline ims/cli/cli.py:78]: why pass nic anymore? this is anyways hardcoded

File: ims/cli/cli.py [hunk 2]
```diff
@@ -93,22 +89,19 @@ def provision(project, node, img, network, nic):
 @cli.command(name='dpro', short_help='Deprovision a node')
 @click.argument(constants.PROJECT_PARAMETER)
 @click.argument(constants.NODE_NAME_PARAMETER)
-@click.argument(constants.NETWORK_PARAMETER)
 @click.argument(constants.NIC_PARAMETER)
-def deprovision(project, node, network, nic):
+def deprovision(project, node, nic):
```
Comment (apoorvemohan) [inline ims/cli/cli.py:93]: again, why do we need nic?

File: ims/einstein/operations.py [hunk 3]
```diff
@@ -212,11 +212,10 @@ def shutdown(self):
 
     # Provisions from HIL and Boots the given node with given image
     @log
-    def provision(self, node_name, img_name, network, nic):
+    def provision(self, node_name, img_name, nic):
```
Comment (apoorvemohan) [inline ims/einstein/operations.py:215]: why nic?

Comment (apoorvemohan) [reply ims/einstein/operations.py:215]: in bmi-config.cfg we can pass it - anyways the nic used for provisioning is different from nic name in HIL

File: ims/picasso/rest.py [hunk 4]
```diff
@@ -83,14 +83,13 @@ def list_images():
 
 @rest_call("/provision/", 'PUT', constants.PROVISION_COMMAND,
            [constants.NODE_NAME_PARAMETER, constants.IMAGE_NAME_PARAMETER,
-            constants.NETWORK_PARAMETER, constants.NIC_PARAMETER])
+            constants.NIC_PARAMETER])
```
Comment (apoorvemohan) [inline ims/picasso/rest.py:86]: lets remove nic?

File: ims/picasso/rest.py [hunk 5]
```diff
@@ -83,14 +83,13 @@ def list_images():
...
 def provision():
     pass
 
 
 @rest_call("/deprovision/", "DELETE", constants.DEPROVISION_COMMAND,
-           [constants.NODE_NAME_PARAMETER, constants.NETWORK_PARAMETER,
-            constants.NIC_PARAMETER])
+           [constants.NODE_NAME_PARAMETER, constants.NIC_PARAMETER])
```
Comment (apoorvemohan) [inline ims/picasso/rest.py:92]: lets remove nic?

File: tests/integration/einstein/test_operations.py [hunk 6]
```diff
@@ -40,20 +39,19 @@ class TestProvision(TestCase):
...
+        self.db.project.insert(PROJECT)
 
         self.good_bmi = BMI(CORRECT_HIL_USERNAME, CORRECT_HIL_PASSWORD,
                             PROJECT)
         self.good_bmi.import_ceph_image(EXIST_IMG_NAME)
 
     def runTest(self):
-        response = self.good_bmi.provision(NODE_NAME, EXIST_IMG_NAME, NETWORK,
-                                           NIC)
+        response = self.good_bmi.provision(NODE_NAME, EXIST_IMG_NAME, NIC)
```
Comment (apoorvemohan) [inline tests/integration/einstein/test_operations.py:49]: remove nic?

File: tests/integration/einstein/test_operations.py [hunk 7]
```diff
@@ -40,20 +39,19 @@ class TestProvision(TestCase):
...
         self.assertEqual(response[constants.STATUS_CODE_KEY], 200)
         time.sleep(constants.HIL_CALL_TIMEOUT)
 
     def tearDown(self):
-        self.good_bmi.deprovision(NODE_NAME, NETWORK, NIC)
+        self.good_bmi.deprovision(NODE_NAME, NIC)
```
Comment (apoorvemohan) [inline tests/integration/einstein/test_operations.py:54]: remove nic?

File: tests/integration/einstein/test_operations.py [hunk 8]
```diff
@@ -68,16 +66,16 @@ class TestDeprovision(TestCase):
...
     def setUp(self):
         self.db = Database()
-        self.db.project.insert(PROJECT, NETWORK)
+        self.db.project.insert(PROJECT)
 
         self.good_bmi = BMI(CORRECT_HIL_USERNAME, CORRECT_HIL_PASSWORD,
                             PROJECT)
         self.good_bmi.import_ceph_image(EXIST_IMG_NAME)
-        self.good_bmi.provision(NODE_NAME, EXIST_IMG_NAME, NETWORK, NIC)
+        self.good_bmi.provision(NODE_NAME, EXIST_IMG_NAME, NIC)
```
Comment (apoorvemohan) [inline tests/integration/einstein/test_operations.py:74]: remove nic?

File: tests/integration/einstein/test_operations.py [hunk 9]
```diff
@@ -68,16 +66,16 @@ class TestDeprovision(TestCase):
...
         time.sleep(constants.HIL_CALL_TIMEOUT)
 
     def runTest(self):
-        response = self.good_bmi.deprovision(NODE_NAME, NETWORK, NIC)
+        response = self.good_bmi.deprovision(NODE_NAME, NIC)
```
Comment (apoorvemohan) [inline tests/integration/einstein/test_operations.py:78]: remove nic?

File: tests/integration/einstein/test_operations.py [hunk 10]
```diff
@@ -95,12 +93,12 @@ class TestCreateSnapshot(TestCase):
...
     def setUp(self):
         self.db = Database()
-        self.db.project.insert(PROJECT, NETWORK)
+        self.db.project.insert(PROJECT)
 
         self.good_bmi = BMI(CORRECT_HIL_USERNAME, CORRECT_HIL_PASSWORD,
                             PROJECT)
         self.good_bmi.import_ceph_image(EXIST_IMG_NAME)
-        self.good_bmi.provision(NODE_NAME, EXIST_IMG_NAME, NETWORK, NIC)
+        self.good_bmi.provision(NODE_NAME, EXIST_IMG_NAME, NIC)
```
Comment (apoorvemohan) [inline tests/integration/einstein/test_operations.py:101]: remove nic?

File: tests/integration/einstein/test_operations.py [hunk 11]
```diff
@@ -121,7 +119,7 @@ def runTest(self):
             fs.get_image(img_id)
 
     def tearDown(self):
-        self.good_bmi.deprovision(NODE_NAME, NETWORK, NIC)
+        self.good_bmi.deprovision(NODE_NAME, NIC)
```
Comment (apoorvemohan) [inline tests/integration/einstein/test_operations.py:122]: remove nic?

File: tests/integration/einstein/test_operations.py [hunk 12]
```diff
@@ -137,12 +135,12 @@ class TestListSnapshots(TestCase):
...
     def setUp(self):
         self.db = Database()
-        self.db.project.insert(PROJECT, NETWORK)
+        self.db.project.insert(PROJECT)
 
         self.good_bmi = BMI(CORRECT_HIL_USERNAME, CORRECT_HIL_PASSWORD,
                             PROJECT)
         self.good_bmi.import_ceph_image(EXIST_IMG_NAME)
-        self.good_bmi.provision(NODE_NAME, EXIST_IMG_NAME, NETWORK, NIC)
+        self.good_bmi.provision(NODE_NAME, EXIST_IMG_NAME, NIC)
```
Comment (apoorvemohan) [inline tests/integration/einstein/test_operations.py:143]: remove nic?

File: tests/integration/einstein/test_operations.py [hunk 13]
```diff
@@ -154,7 +152,7 @@ def runTest(self):
                          NEW_SNAP_NAME)
 
     def tearDown(self):
-        self.good_bmi.deprovision(NODE_NAME, NETWORK, NIC)
+        self.good_bmi.deprovision(NODE_NAME, NIC)
```
Comment (apoorvemohan) [inline tests/integration/einstein/test_operations.py:155]: remove nic?

File: tests/unit/picasso/test_rest.py [hunk 14]
```diff
@@ -98,15 +97,14 @@ def runTest(self):
...
-                constants.NETWORK_PARAMETER: NETWORK,
                 constants.NIC_PARAMETER: NIC}
         res = requests.put(PICASSO_URL + "provision/", data=data,
                            auth=(CORRECT_HIL_USERNAME, CORRECT_HIL_PASSWORD))
         self.assertEqual(res.status_code, 200)
         time.sleep(constants.HIL_CALL_TIMEOUT)
 
     def tearDown(self):
-        self.good_bmi.deprovision(NODE_NAME, NETWORK, NIC)
+        self.good_bmi.deprovision(NODE_NAME, NIC)
```
Comment (apoorvemohan) [inline tests/unit/picasso/test_rest.py:107]: remove nic?

File: tests/unit/picasso/test_rest.py [hunk 15]
```diff
@@ -122,17 +120,16 @@ class TestDeprovision(TestCase):
     @trace
     def setUp(self):
         self.db = Database()
-        self.db.project.insert(PROJECT, NETWORK)
+        self.db.project.insert(PROJECT)
         self.good_bmi = BMI(CORRECT_HIL_USERNAME, CORRECT_HIL_PASSWORD,
                             PROJECT)
         self.good_bmi.import_ceph_image(EXIST_IMG_NAME)
-        self.good_bmi.provision(NODE_NAME, EXIST_IMG_NAME, NETWORK, NIC)
+        self.good_bmi.provision(NODE_NAME, EXIST_IMG_NAME, NIC)
```
Comment (apoorvemohan) [inline tests/unit/picasso/test_rest.py:127]: remove nic?

File: tests/unit/picasso/test_rest.py [hunk 16]
```diff
@@ -156,11 +153,11 @@ class TestCreateSnapshot(TestCase):
...
```
Comment (apoorvemohan) [inline tests/unit/picasso/test_rest.py:160]: remove nic?

File: tests/unit/picasso/test_rest.py [hunk 17]
```diff
@@ -185,7 +182,7 @@ def runTest(self):
             fs.get_image(img_id)
 
     def tearDown(self):
-        self.good_bmi.deprovision(NODE_NAME, NETWORK, NIC)
+        self.good_bmi.deprovision(NODE_NAME, NIC)
```
Comment (apoorvemohan) [inline tests/unit/picasso/test_rest.py:185]: remove nic?

File: tests/unit/picasso/test_rest.py [hunk 18]
```diff
@@ -202,11 +199,11 @@ class TestListSnapshots(TestCase):
     @trace
     def setUp(self):
         self.db = Database()
-        self.db.project.insert(PROJECT, NETWORK)
+        self.db.project.insert(PROJECT)
         self.good_bmi = BMI(CORRECT_HIL_USERNAME, CORRECT_HIL_PASSWORD,
                             PROJECT)
         self.good_bmi.import_ceph_image(EXIST_IMG_NAME)
-        self.good_bmi.provision(NODE_NAME, EXIST_IMG_NAME, NETWORK, NIC)
+        self.good_bmi.provision(NODE_NAME, EXIST_IMG_NAME, NIC)
```
Comment (apoorvemohan) [inline tests/unit/picasso/test_rest.py:206]: remove nic?

File: tests/unit/picasso/test_rest.py [hunk 19]
```diff
@@ -222,7 +219,7 @@ def runTest(self):
         self.assertEqual(js[0][0], NEW_SNAP_NAME)
 
     def tearDown(self):
-        self.good_bmi.deprovision(NODE_NAME, NETWORK, NIC)
+        self.good_bmi.deprovision(NODE_NAME, NIC)
```
Comment (apoorvemohan) [inline tests/unit/picasso/test_rest.py:222]: remove nic?

**Review by ianballou — CHANGES_REQUESTED** [review]
I'm gonna retract the +1 till we figure out the DB issues here: #174 

Edit: still not certain that this PR is the cause, but it's good to be safe

**Comment by naved001** [comment]
The CI was passing for this PR earlier, and now something else seems to be failing (some compilation error).

**Comment by naved001** [comment]
@ianballou  fixed the CI.

**Comment by ianballou** [comment]
@naved001 were you able to test that added images still get related to the project?  I was worried that might've slipped through the CI somehow when I tested this code before. If that's all good, then I'm cool with this.

**Review by ianballou — APPROVED** [review]

**Comment by naved001** [comment]
@apoorvemohan I can remove the nic from the dpro call if you want, but I will add it back again in the other PR for cleanup (to delete the boot configuration file).

**Review by apoorvemohan — APPROVED** [review]

**Pull request merged**